### DISCUSSION TITLE
No Data since December 6th

### DISCUSSION THREAD

**Discussion by tom-davies** [BottlecapDave/HomeAssistant-OctopusEnergy#629] [General]
I've not had any consumption or price data updates since December 6th. Sometimes I've had dips in the past due to meters not communicating with OE etc, but everything is showing up in their dashboard.

I'm seeing the following lines repeatedly in the logs:

``` 
2023-12-12 20:33:46.596 DEBUG (MainThread) [custom_components.octopus_energy.coordinators.previous_consumption_and_rates] Finished fetching REDACTED_REDACTED_previous_consumption_and_rates data in 0.000 seconds (success: True)
2023-12-12 20:33:46.597 DEBUG (MainThread) [custom_components.octopus_energy.coordinators.previous_consumption_and_rates] Finished fetching REDACTED_REDACTED_previous_consumption_and_rates data in 0.000 seconds (success: True)
```

It looks like any fetch at all is complete in 0.000 seconds, so maybe the culprit is to be found there?

```curl```ing the consumption from the device that HA is running on works fine, but definitely takes longer than 0 seconds (```time``` has it at 2.401 seconds).

Rebooting hasn't solved it, nor has updating to 9.2.0-beta1.

**Comment by BottlecapDave** [comment]
This is normal, as the coordinator runs every minute, but doesn't necessarily do any requests every minute. If you're on the beta, then it will display a warning stating the next refresh if it fails to download data. When looking at the dashboard are you looking on the web or in the app? Do you have a home mini? When curling the request, do you include the necessary filters for the consumption data, as today I've noticed the API is broken and returning data with no filters, but not returning any data when filters are applied.

**Comment by tom-davies** [comment]
I've checked the dashboard on web and app, right now the app is showing data, but the website isn't.

I don't have a home mini.

I'm not sure how to apply the filters when curling the request, i just use ```curl -u "api_key:" "https://api.octopus.energy/v1/electricity-meter-points/REDACTED/meters/REDACTED/consumption/"```

**Reply by BottlecapDave** [reply]
Filters can be applied via query parameters. The OE docs https://docs.octopus.energy/rest/guides/endpoints#api-consumption-endpoints have an example of applying the filters.

However, if the data isn't available on the website there's a good chance there's something wrong with either the API or your OE account. The former had issues yesterday but I've been told have since been fixed. If it's still not reporting data, you might need to contact OE.

**Reply by tom-davies** [reply]
Looks like I can get data with filters now, and the website is showing the data too. I'll get in touch with OE, see what they say and report back.

**Comment by tom-davies** [comment]
It looks like I sorted it by uninstalling, reinstalling 9.2.0-beta2 and calling the ```octopus_energy.refresh_previous_consumption_data``` service.
